index.js

```diff
@@ -1,39 +1,171 @@
+There is a door here. A keypad sits on the handle.
+On the door is a handwritten sign.
+
+`,
+		readSign: `
+
+The sign says "Welcome to Burlington Code Academy! Come on 
+up to the third floor. If the door is locked, use the code
+12345."
+
+`
+	},
+	"foyer": {
+		canChangeTo: ["182main", "stairs"],
+		name: "Foyer",
+		welcomeMessage: `
+You are in a foyer. Or maybe it's an antechamber. Or a 
+vestibule. Or an entryway. Or an atrium. Or a narthex.
+But let's forget all that fancy flatlander vocabulary,
+and just call it a foyer. In Vermont, this is pronounced
+"FO-ee-yurr".
+
+A copy of Seven Days lies in a corner.
+
+`,
+		readPaper: `
+You pick up the paper and leaf through it looking for comics 
+and ignoring the articles, just like everybody else does.
+
+`,
+		takePaper: `
+You are carrying:
+A copy of Seven Days, Vermont's Alt-Weekly
+
+`,
+		dropPaper: `
+Paper has been dropped.
+
+`
+	},
+	"stairs": { canChangeTo: ["classroom", "foyer"], name: "Stairs" },
+	"mr.mikes": { canChangeTo: ["182main"] },
+	"muddys": { canChangeTo: ["182main"] },
+	"classroom": { canChangeTo: ["stairs"],
+		welcomeMessage: `
+		
+You made it to class but no one has coffee including yourself. 
+So program explodes enter exit and leave program now so you can fix it!
+
+` }
 };
 
-let currentState = "green";
+/*let character = { Character Object. Still not sure exactly how I want to implement.
+	"inventory": {
+
+	},
+	"name": 'agent Z',
+	"appetite": {},
+
+}*/ 
+
+let currentState = "182main";
+
+function enterState(newState) {//  State Machine for room transition rules. 
+let validTransitions = states[currentState].canChangeTo; //Do I need another seperate function for inventory?
+	if (validTransitions.includes(newState)) {
+		currentState = newState;
+	} else {
+		throw "Invalid state transition attempted - from " +
+			currentState +
+			" to " +
+			newState;
+	}
+}
 
-function enterState(newState) {
-  let validTransitions = states[currentState].canChangeTo;
-  if (validTransitions.includes(newState)) {
-    currentState = newState;
-  } else {
-    throw 'Invalid state transition attempted - from ' + currentState + ' to ' + newState;
-  }
+function checkPinCode(code) {
+	code = answer.includes("enter code", 0);
+	if (code === "true") {
+		console.log("Please enter code now!");
+	} else {
+		console.log("Try again");
+	}
 }
 
 start();
 
-async function start() {
-  const welcomeMessage = `182 Main St.
-You are standing on Main Street between Church and South Winooski.
-There is a door here. A keypad sits on the handle.
-On the door is a handwritten sign.`;
-  let answer = await ask(welcomeMessage);
-  console.log('Now write your code to make this work!');
-  process.exit();
+async function start() {                  // This whole loop is controlling the story depending on things typed.
+	console.log(states[currentState].welcomeMessage);
+	let answer = await ask(">_");
+	answer = answer.toLowerCase();
+	while (answer !== "exit") {
```

Comment:
This loop gets you through the backlog as long as the user follows the stories exactly, which a real world user will almost never do. It would probably be better to split up the user input and return it as an array. You could then call specific methods based on the commands entered and the room you're in which would make your project more scale-able

Reply:
I get what you are saying definitely hard for me to visualize how to use it given we haven't seen how to do this more than a couple times.